### PR TITLE
use crypto/rand for randRead

### PR DESCRIPTION
**Pull request by cmiller01**
* randRead is needed to generate nonce and keys, which should be from crypto/rand
vs math/rand

### CODE REVIEW

File: go/util/crypto.go [hunk 1]
```diff
@@ -101,7 +102,7 @@ func GenerateSaltsAndKey(token types.Token, secret string) types.SaltsAndKey {
 	hashable := sha3.Sum512([]byte(tokenStr))
 	seed := append([]byte(tokenStr), hashable[:]...)
 	seedInt := binary.BigEndian.Uint64(seed)
-	random := rand.New(rand.NewSource(int64(seedInt)))
+	random := mathrand.New(mathrand.NewSource(int64(seedInt)))
 
 	return types.SaltsAndKey{
 		Sp: GetRandomStringOfLength(keySize, random),
```
Comment (cmiller01) [inline go/util/crypto.go:108]: there's some interesting things going on here, maybe it's intended, but you're re-using the random reader on each `GetRandomStringOfLength` call below, so the ordering of these objects in the struct matters...maybe this is intended, but wanted to check

**Comment by freeqaz** [comment]
Tests all passed, but the last line of the CI script fails to push to Slack. I'm going to land this and fix that issue in another branch. Thanks for the PR!

File: go/util/crypto.go (hunk 1 above)
Comment (freeqaz) [reply go/util/crypto.go:108]: Yep, that's intended. It's how we deterministically derive these values.

**Review by freeqaz — APPROVED** [review]
LGTM!

**Pull request merged**